cmd/update.go

```diff
@@ -0,0 +1,137 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"runtime"
+	"sort"
+	"sync"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/talos-systems/bldr/internal/pkg/solver"
+	"github.com/talos-systems/bldr/internal/pkg/update"
+)
+
+type pkgInfo struct {
+	file   string
+	source string
+}
+
+type updateInfo struct {
+	file string
+	*update.LatestInfo
+}
+
+var updateCmdFlag struct {
+	all bool
+	dry bool
+}
+
+// updateCmd represents the `update` command.
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update pkgs",
+	Run: func(cmd *cobra.Command, args []string) {
+		if !updateCmdFlag.dry {
+			log.Fatal("Real update is not implemented yet; pass `--dry` flag.")
+		}
+
+		loader := solver.FilesystemPackageLoader{
+			Root:    pkgRoot,
+			Context: options.GetVariables(),
+		}
+
+		packages, err := solver.NewPackages(&loader)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		l := log.New(log.Writer(), "[update] ", log.Flags())
+		if !debug {
+			l.SetOutput(ioutil.Discard)
+		}
+
+		concurrency := runtime.GOMAXPROCS(-1)
+		var wg sync.WaitGroup
```

Comment:
Should we use an error group? Do we want to treat a failed update is an error?

Reply:
At the moment, all errors are treated as warnings and logged and ignored below. Once more cases are covered, and the distinction between hard errors and warnings is more clear, this code will be updated to use ErrGroup.